src/App.tsx

```diff
@@ -15,36 +16,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = 'default',
+}
+
+const handleSort = (
+  type: SortType,
+  setGoods: React.Dispatch<React.SetStateAction<string[]>>,
+  isReversed: boolean,
+) => {
+  let sortedGoods = [...goodsFromServer];
+
+  switch (type) {
+    case SortType.ALPHABET:
+      sortedGoods.sort();
+      break;
+    case SortType.LENGTH:
+      sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      sortedGoods = [...goodsFromServer];
```

Comment:
You have already created a copy, so you don't need to do it one more time

Reply:
Not fixed!